The job **`x86_64-gnu-llvm-17`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/10292658521/job/28487491682) [(plain)](https://github.com/rust-lang/rust/commit/b9033c924a4f50139fcbe671033632447379d892/checks/28487491682/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
------
 > importing cache manifest from ghcr.io/rust-lang/rust-ci-cache:3aacb9c90579defe09351ac5e8ee504359f8054da6326ff19038f1b7c90e3cb2aafe33685c6d9b76ee8d2ccbd187ca80c46ab5380485abdd8c0ce7d69cd8d8fd:
------
##[endgroup]
Setting extra environment values for docker:  --env ENABLE_GCC_CODEGEN=1 --env GCC_EXEC_PREFIX=/usr/lib/gcc/
[CI_JOB_NAME=x86_64-gnu-llvm-17]
---
sccache: Starting the server...
##[group]Configure the build
configure: processing command line
configure: 
configure: build.configure-args := ['--build=x86_64-unknown-linux-gnu', '--llvm-root=/usr/lib/llvm-17', '--enable-llvm-link-shared', '--set', 'rust.thin-lto-import-instr-limit=10', '--set', 'change-id=99999999', '--enable-verbose-configure', '--enable-sccache', '--disable-manage-submodules', '--enable-locked-deps', '--enable-cargo-native-static', '--set', 'rust.codegen-units-std=1', '--set', 'dist.compression-profile=balanced', '--dist-compression-formats=xz', '--set', 'rust.lld=false', '--disable-dist-src', '--release-channel=nightly', '--enable-debug-assertions', '--enable-overflow-checks', '--enable-llvm-assertions', '--set', 'rust.verify-llvm-ir', '--set', 'rust.codegen-backends=llvm,cranelift,gcc', '--set', 'llvm.static-libstdcpp', '--enable-new-symbol-mangling']
configure: target.x86_64-unknown-linux-gnu.llvm-config := /usr/lib/llvm-17/bin/llvm-config
configure: llvm.link-shared     := True
configure: rust.thin-lto-import-instr-limit := 10
configure: change-id            := 99999999
---

---- [ui] tests/ui/type/pattern_types/feature-gate-pattern_types.rs stdout ----
diff of stderr:

4 LL | type NonNullU32 = pattern_type!(u32 is 1..);
6    |
+    = note: see issue #123646 <https://github.com/rust-lang/rust/issues/123646> for more information
7    = help: add `#![feature(core_pattern_type)]` to the crate attributes to enable
7    = help: add `#![feature(core_pattern_type)]` to the crate attributes to enable
8    = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date


13 LL | type Percent = pattern_type!(u32 is 0..=100);
15    |
+    = note: see issue #123646 <https://github.com/rust-lang/rust/issues/123646> for more information
16    = help: add `#![feature(core_pattern_type)]` to the crate attributes to enable
16    = help: add `#![feature(core_pattern_type)]` to the crate attributes to enable
17    = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date


22 LL | type Negative = pattern_type!(i32 is ..=0);
24    |
+    = note: see issue #123646 <https://github.com/rust-lang/rust/issues/123646> for more information
25    = help: add `#![feature(core_pattern_type)]` to the crate attributes to enable
25    = help: add `#![feature(core_pattern_type)]` to the crate attributes to enable
26    = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date


31 LL | type Positive = pattern_type!(i32 is 0..);
33    |
+    = note: see issue #123646 <https://github.com/rust-lang/rust/issues/123646> for more information
34    = help: add `#![feature(core_pattern_type)]` to the crate attributes to enable
34    = help: add `#![feature(core_pattern_type)]` to the crate attributes to enable
35    = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date


40 LL | type Always = pattern_type!(Option<u32> is Some(_));
42    |
+    = note: see issue #123646 <https://github.com/rust-lang/rust/issues/123646> for more information
43    = help: add `#![feature(core_pattern_type)]` to the crate attributes to enable
43    = help: add `#![feature(core_pattern_type)]` to the crate attributes to enable
44    = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date


The actual stderr differed from the expected stderr.
Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/type/pattern_types/feature-gate-pattern_types/feature-gate-pattern_types.stderr
Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/type/pattern_types/feature-gate-pattern_types/feature-gate-pattern_types.stderr
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args type/pattern_types/feature-gate-pattern_types.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/type/pattern_types/feature-gate-pattern_types.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/type/pattern_types/feature-gate-pattern_types" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/type/pattern_types/feature-gate-pattern_types/auxiliary" "-Zno-analysis"
--- stderr -------------------------------
error[E0658]: use of unstable library feature 'core_pattern_type'
##[error]  --> /checkout/tests/ui/type/pattern_types/feature-gate-pattern_types.rs:5:19
   |
   |
LL | type NonNullU32 = pattern_type!(u32 is 1..);
   |
   = note: see issue #123646 <https://github.com/rust-lang/rust/issues/123646> for more information
   = help: add `#![feature(core_pattern_type)]` to the crate attributes to enable
   = help: add `#![feature(core_pattern_type)]` to the crate attributes to enable
   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date
error[E0658]: use of unstable library feature 'core_pattern_type'
##[error]  --> /checkout/tests/ui/type/pattern_types/feature-gate-pattern_types.rs:7:16
   |
   |
LL | type Percent = pattern_type!(u32 is 0..=100);
   |
   = note: see issue #123646 <https://github.com/rust-lang/rust/issues/123646> for more information
   = help: add `#![feature(core_pattern_type)]` to the crate attributes to enable
   = help: add `#![feature(core_pattern_type)]` to the crate attributes to enable
   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date
error[E0658]: use of unstable library feature 'core_pattern_type'
##[error]  --> /checkout/tests/ui/type/pattern_types/feature-gate-pattern_types.rs:9:17
   |
   |
LL | type Negative = pattern_type!(i32 is ..=0);
   |
   = note: see issue #123646 <https://github.com/rust-lang/rust/issues/123646> for more information
   = help: add `#![feature(core_pattern_type)]` to the crate attributes to enable
   = help: add `#![feature(core_pattern_type)]` to the crate attributes to enable
   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date
error[E0658]: use of unstable library feature 'core_pattern_type'
##[error]  --> /checkout/tests/ui/type/pattern_types/feature-gate-pattern_types.rs:11:17
   |
   |
LL | type Positive = pattern_type!(i32 is 0..);
   |
   = note: see issue #123646 <https://github.com/rust-lang/rust/issues/123646> for more information
   = help: add `#![feature(core_pattern_type)]` to the crate attributes to enable
   = help: add `#![feature(core_pattern_type)]` to the crate attributes to enable
   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date
error[E0658]: use of unstable library feature 'core_pattern_type'
##[error]  --> /checkout/tests/ui/type/pattern_types/feature-gate-pattern_types.rs:13:15
   |
   |
LL | type Always = pattern_type!(Option<u32> is Some(_));
   |
   = note: see issue #123646 <https://github.com/rust-lang/rust/issues/123646> for more information
   = help: add `#![feature(core_pattern_type)]` to the crate attributes to enable
   = help: add `#![feature(core_pattern_type)]` to the crate attributes to enable
   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date
error: aborting due to 5 previous errors

For more information about this error, try `rustc --explain E0658`.
------------------------------------------
```

</details>

--- a/library/core/src/lib.rs
+++ b/library/core/src/lib.rs
@@ -391,7 +391,7 @@ pub mod net;
 pub mod option;
 pub mod panic;
 pub mod panicking;
-#[unstable(feature = "core_pattern_types", issue = "none")]
+#[unstable(feature = "core_pattern_types", issue = "123646")]
 pub mod pat;
 pub mod pin;
 #[unstable(feature = "new_range_api", issue = "125687")]

--- a/library/core/src/pat.rs
+++ b/library/core/src/pat.rs
@@ -6,7 +6,7 @@
 /// ```
 #[macro_export]
 #[rustc_builtin_macro(pattern_type)]
-#[unstable(feature = "core_pattern_type", issue = "none")]
+#[unstable(feature = "core_pattern_type", issue = "123646")]
 macro_rules! pattern_type {
     ($($arg:tt)*) => {
         /* compiler built-in */

--- a/library/std/src/lib.rs
+++ b/library/std/src/lib.rs
@@ -586,7 +586,7 @@ pub mod net;
 pub mod num;
 pub mod os;
 pub mod panic;
-#[unstable(feature = "core_pattern_types", issue = "none")]
+#[unstable(feature = "core_pattern_types", issue = "123646")]
 pub mod pat;
 pub mod path;
 #[unstable(feature = "anonymous_pipe", issue = "127154")]

--- a/tests/ui/type/pattern_types/feature-gate-pattern_types.stderr
+++ b/tests/ui/type/pattern_types/feature-gate-pattern_types.stderr
@@ -4,6 +4,7 @@ error[E0658]: use of unstable library feature 'core_pattern_type'
 LL | type NonNullU32 = pattern_type!(u32 is 1..);
    |                   ^^^^^^^^^^^^
    |
+   = note: see issue #123646 <https://github.com/rust-lang/rust/issues/123646> for more information
    = help: add `#![feature(core_pattern_type)]` to the crate attributes to enable
    = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date
 
@@ -13,6 +14,7 @@ error[E0658]: use of unstable library feature 'core_pattern_type'
 LL | type Percent = pattern_type!(u32 is 0..=100);
    |                ^^^^^^^^^^^^
    |
+   = note: see issue #123646 <https://github.com/rust-lang/rust/issues/123646> for more information
    = help: add `#![feature(core_pattern_type)]` to the crate attributes to enable
    = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date
 
@@ -22,6 +24,7 @@ error[E0658]: use of unstable library feature 'core_pattern_type'
 LL | type Negative = pattern_type!(i32 is ..=0);
    |                 ^^^^^^^^^^^^
    |
+   = note: see issue #123646 <https://github.com/rust-lang/rust/issues/123646> for more information
    = help: add `#![feature(core_pattern_type)]` to the crate attributes to enable
    = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date
 
@@ -31,6 +34,7 @@ error[E0658]: use of unstable library feature 'core_pattern_type'
 LL | type Positive = pattern_type!(i32 is 0..);
    |                 ^^^^^^^^^^^^
    |
+   = note: see issue #123646 <https://github.com/rust-lang/rust/issues/123646> for more information
    = help: add `#![feature(core_pattern_type)]` to the crate attributes to enable
    = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date
 
@@ -40,6 +44,7 @@ error[E0658]: use of unstable library feature 'core_pattern_type'
 LL | type Always = pattern_type!(Option<u32> is Some(_));
    |               ^^^^^^^^^^^^
    |
+   = note: see issue #123646 <https://github.com/rust-lang/rust/issues/123646> for more information
    = help: add `#![feature(core_pattern_type)]` to the crate attributes to enable
    = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date
 

